File: ms_cinema/infrastructure/driven-adapters/r2dbc-postgresql/src/main/java/co/com/luisgomez29/r2dbc/genre/data/GenreMapper.java
package co.com.luisgomez29.r2dbc.genre.data;

import co.com.luisgomez29.model.genre.Genre;
import org.mapstruct.Mapper;


@Mapper(componentModel = "spring")
public interface GenreMapper {

    Genre toEntity(GenreData genreData);

    GenreData toData(Genre genre);
}

File: ms_cinema_kt/infrastructure/driven-adapters/r2dbc-postgresql/src/main/kotlin/co/com/luisgomez29/genre/data/GenreMapper.kt
package co.com.luisgomez29.genre.data

import co.com.luisgomez29.model.genre.Genre
import org.mapstruct.Mapper

@Mapper(componentModel = "spring")
interface GenreMapper {
    fun toEntity(genreData: GenreData): Genre
    fun toData(genre: Genre): GenreData
}
